partner_preview_scripts/encore_parser.py

```diff
@@ -0,0 +1,410 @@
+        logging.info(f'Number cell lines in the gemini dataset: {gemini_df.select("cellLineName").distinct().count()} rows')
+
+        # Reading bliss data: <= not for now.
+        # bliss_file = f'{self.data_folder}/{blissFile}'
+        # bliss_df = self.get_bliss_data(bliss_file)
+
+        # Merging lfc + gemini:
+        merged_dataset = (
+            lfc_df
+
+            # Data is joined by the gene-pair and cell line:
+            # .join(bliss_df, how='inner', on=['id', 'cellLineName'])
+            .join(gemini_df, how='inner', on=['id', 'cellLineName'])
+
+            # Applying filters on logFoldChange + interaction p-value thresholds:
+            .filter(
+                (col('phenotypicConsequencePValue') <= self.log_fold_change_cutoff_p_val) &
+                (col('phenotypicConsequenceFDR') <= self.log_fold_change_cutoff_FDR) &
+                (col('geneticInteractionPValue') <= self.interaction_cutoff_p_val)
+            )
+
+            # Cleaning the cell line annotation:
+            .withColumn('cellId', split(col('cellLineName'), '_').getItem(0))
+
+            # Joining with cell passport data containing diseaseCellLines and biomarkers info:
+            .join(
+                self.cell_passport_df.select(col('id').alias('cellId'), 'diseaseCellLines', 'biomarkerList'),
+                on='cellId', how='left'
+            )
+            .persist()
+        )
+        logging.info(f'Number of gene paris in the merged dataset: {merged_dataset.select("id").count()} rows')
+        logging.info(f'Number cell lines in the merged dataset: {merged_dataset.select("cellLineName").distinct().count()} rows')
+
+        evidence_df = (
+            merged_dataset
+
+            .withColumn('diseaseCellLines', array(col('diseaseCellLines')))
+
+            # Parsing/exploding gene names and target roles:
+            .withColumn('id', self.parse_targets(col('id'), col('Note1')))
+            .select('*', explode(col('id')).alias('genes'))
+            .select('*', col('genes.*'))
+
+            # Adding some literals specific for this type of evidence:
+            .withColumn('datatypeId', lit('ot_partner'))
+            .withColumn('datasourceId', lit('encore'))
+            .withColumn('projectId', lit('OTAR2062'))
+            .withColumn('projectDescription', lit('Encore project'))  ### TODO - fix this!
+            .withColumn('geneInteractionType', lit('cooperative'))
+
+            # Adding disease information:
+            .withColumn('diseaseFromSourceMappedId', lit(disease_from_source_mapped_id))
+            .withColumn('diseaseFromSource', lit(disease_from_source))
+            .withColumn('releaseVersion', lit(self.release_version))
+            .withColumn('releaseDate', lit(self.release_date))
+
+            # Removing unused columns:
+            .drop(*['cellLineName', 'cellId', 'Note1', 'Note2', 'id', 'genes'])
```

Comment:
This is purely a matter of personal preference, but in such cases, I prefer to have a "select everything you need" approach rather than "drop everything you don't need". This also has two added bonuses:
1. Helps with readability, because at the end of a large dataframe constructions you can see which fields are being kept
2. Ensures the stable sort order of all the columns

Reply:
I'm addressing this issue now.